test/extern/fnPtrs/improveErrorForFcfMismatch.bad

```diff
@@ -1,4 +1,4 @@
-improveErrorForFcfMismatch.chpl:13: error: unresolved call 'bar(shared chpl__fcf_type_void_void)'
```

Comment:
Do we want this? (`chpl__` to `proc()`? Seems like it's losing some descriptiveness).

Reply:
The previous type was the underlying generated type, now it is user facing syntax for a function type.